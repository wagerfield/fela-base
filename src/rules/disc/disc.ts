export interface DiscRuleProps {
  color?: string
  size?: number
}

export default ({
  color = "black",
  size = 40
}: DiscRuleProps = {}) => ({
  borderRadius: size ? size * 0.5 : null,
  backgroundColor: color,
  boxSizing: "border-box",
  size
})
