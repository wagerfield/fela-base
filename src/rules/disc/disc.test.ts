import disc from "./disc"

describe("disc(props)", () => {
  it("renders expected snapshots", () => {
    expect(disc()).toMatchSnapshot()
    expect(
      disc({
        size: 100,
        color: "red"
      })
    ).toMatchSnapshot()
  })
})
