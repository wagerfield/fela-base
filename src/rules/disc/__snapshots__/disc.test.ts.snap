// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`disc(props) renders expected snapshots 1`] = `
Object {
  "backgroundColor": "black",
  "borderRadius": 20,
  "boxSizing": "border-box",
  "size": 40,
}
`;

exports[`disc(props) renders expected snapshots 2`] = `
Object {
  "backgroundColor": "red",
  "borderRadius": 50,
  "boxSizing": "border-box",
  "size": 100,
}
`;
